hotsos/core/plugintools.py

```diff
@@ -381,13 +394,52 @@ def summary(self):
         """
         return None
 
+    @classmethod
+    def get_summary_functions(cls):
+        """Return a list of summary functions defined in the current class
+        and all the derived classes.
+
+        Summary functions are the functions decorated with @summary_function
+        decorator.
+        """
+
+        result = {}
+        # Iterate through all attributes of BaseClass
+        for name in dir(cls):
+            # Get the attribute from the current class (cls)
+            attr = getattr(cls, name)
+            # Check if it's a function and has the specified attribute
+            if callable(attr) and hasattr(attr, '__summary_elem_name'):
+                name = getattr(attr, '__summary_elem_name')
+                index = getattr(attr, '__summary_elem_index')
+                log.debug("Registered summary function %s with idx %d",
+                          name, index)
+                result[name] = [attr.__name__, index]
+        return result
+
     @property
     def output(self):
         _output = {}
         if self.summary:
             for key, data in self.summary.items():
                 _output[key] = SummaryEntry(data, 0)
 
+        # Discover the summary functions
+        implicit_summary_fns = self.get_summary_functions()
+
+        if implicit_summary_fns:
```

Comment:
I'm not sure this sorting by index is necessary here as the sorting is done later on when the output is assembled.

Reply:
I did this because unit tests were failing otherwise. I can modify the unit tests if it's ok for you.

Reply:
hmm interesting,I can look into that later. Lets leave as is then.